src/wasm.h

```diff
@@ -988,10 +988,11 @@ class Memory {
     Segment(size_t offset, const char *data, size_t size) : offset(offset), data(data), size(size) {}
   };
 
+  bool read;
```

Comment:
i think i understand this now, but see comment above about another approach - i don't think we should modify this here.